src/main/java/com/mparticle/kits/ButtonKit.java

```diff
@@ -45,27 +45,31 @@ public ButtonKit getInstance() {
             throw new IllegalArgumentException(
                     "No Button application ID provided, can't initialize kit.");
         }
-        ButtonMerchant.configure(ctx, applicationId);
+        ButtonMerchant.configure(getApplicationContext(), applicationId);
+        logDebug("Button Attribution Token: %s",
+                ButtonMerchant.getAttributionToken(getApplicationContext()));
 
-        ButtonMerchant.addAttributionTokenListener(ctx,
+        ButtonMerchant.addAttributionTokenListener(getApplicationContext(),
                 new ButtonMerchant.AttributionTokenListener() {
                     @Override
-                    public void onAttributionTokenChanged(@NonNull String s) {
+                    public void onAttributionTokenChanged(@NonNull final String s) {
                         logDebug("Refreshed Button Attribution Token: %s", s);
                     }
                 });
 
-        ButtonMerchant.handlePostInstallIntent(ctx, new PostInstallIntentListener() {
-            @Override
-            public void onResult(@Nullable Intent intent, @Nullable Throwable throwable) {
-                PackageManager pm = ctx.getPackageManager();
-                if (intent != null && intent.resolveActivity(pm) != null) {
-                    ctx.startActivity(intent);
-                } else if (throwable != null) {
-                    logError("Error checking post install intent", throwable);
-                }
-            }
-        });
+        ButtonMerchant.handlePostInstallIntent(getApplicationContext(),
+                new PostInstallIntentListener() {
+                    @Override
+                    public void onResult(@Nullable Intent intent, @Nullable Throwable throwable) {
+                        PackageManager pm = getApplicationContext().getPackageManager();
+                        if (intent != null && intent.resolveActivity(pm) != null) {
+                            logDebug("Handling post-install intent for %s", intent.toString());
+                            getApplicationContext().startActivity(intent);
+                        } else if (throwable != null) {
+                            logError("Error checking post install intent", throwable);
+                        }
```

Comment:
what about the case of if there isn't an intent  and there isn't a throwable?

Reply:
If neither the intent nor the throwable are available then we'd have to ignore the "result". Though since it is a possibility, I could log an error.